### ISSUE TITLE
Is the network embedding loss included? 

### ISSUE THREAD

**Issue by mickeystroller** [facebookresearch/poincare-embeddings#11]
Hi, 

I am trying to figure out what the exact loss used in for the network embeddings (Section 4.2 in the paper). However, after reading the code, I find only one loss is used (defined in the SNEmbedding class) and it is a CrossEntropyLoss corresponding to the Eq. (5) in the paper. I wonder whether I miss anything or this repo just doesn't contain the code for network embedding part? If it's the latter case, would you mind revealing the exact loss used for network embedding? My current guess for that loss will be the one that replaces e^{-d(u,v)} with P((u,v)=1|\Theta) in Eq. (5). Am I right? 

P.S.: I am referring to the [NIPS version](https://papers.nips.cc/paper/7213-poincare-embeddings-for-learning-hierarchical-representations.pdf) of paper. 

Thanks.

**Comment by ArgentumBlack** [comment]
I have similar questions. The codes seem to apply cross-entopy as loss function, which is different with Eq.(5). But compared with Eq.(6), the parameters r and t seem to be missing.  I wonder whether Eq.(5) or Eq.(6) is used.

Thanks a lot!

**Comment by rem2016** [comment]
The cross-entropy is exactly Eq.(6). You can compare the equation with 
 https://pytorch.org/docs/master/nn.html#torch.nn.CrossEntropyLoss
